### PR TITLE
Add missing params to estimator_params in sub-classes

### LINKED ISSUE #821: Add subclass-specific params to estimator_params list
The BaseEstimator get_params() method returns all params that were defined in the estimator's estimator_params attribute. However, some subclasses do not add their specific params to this list, therefore when calling get_params() on them, only the base classes parameters are returned. For example in TensorFlowV2Classifier, params such as nb_classes, input_shape, train_step, reduce_labels, etc. should also be added.

### PR DESCRIPTION
**Pull request by abigailgold**
# Description

All sub-classes of BaseEstimator should add to the estimator_params field their specific parameters.

Fixes #821

## Type of change

Please check all relevant options.

- [ ] Improvement (non-breaking)
- [x] Bug fix (non-breaking)
- [ ] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Testing

Please describe the tests that you ran to verify your changes. Consider listing any relevant details of your test configuration.

Unit tests for changed classes.

**Test Configuration**:
- OS: MacOS 10.15.7
- Python version 3.7
- ART version or commit number
- TensorFlow / Keras / PyTorch / MXNet version

# Checklist

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: art/estimators/object_detection/tensorflow_faster_rcnn.py [hunk 1]
```diff
@@ -205,6 +211,9 @@ def input_shape(self) -> Tuple[int, ...]:
         """
         return self._input_shape  # type: ignore
 
+    #         "filename",
+    #         "url",
+
```
Comment (beat-buesser) [inline art/estimators/object_detection/tensorflow_faster_rcnn.py:216]: Can we remove these line?

Comment (abigailgold) [reply art/estimators/object_detection/tensorflow_faster_rcnn.py:216]: Yes, of course. I will push a fix.

**Review by beat-buesser — CHANGES_REQUESTED** [review]
Hi @abigailgold Thank you very much for the pull request!

It looks like in classes inheriting from multiple base classes only one base class is used to update `estimator_params`. Would it be possible to include the arguments of all base classes, including the mixins, in `estimator_params`?

Should we update the codes and replace the calls to `self._variable` with calls to the corresponding new property `self.variable`?

**Comment by abigailgold** [comment]
Hi @beat-buesser in most mixin classes I did not see any params, and if there were I just included them directly in the sub-class params (such as in the case of channels_first in NeuralNetworkMixin). If you prefer I can move that into the mixin class and have all sub-classes do: estimator_params = PyTorchEstimator.estimator_params + NeuralNetworkMixin.estimator_params + ...). Is this what you meant?

Regarding the calls to self._variable, I think both options are ok, so whatever you think is best.

**Comment by beat-buesser** [comment]
@abigailgold Great, I didn't notice. I think `estimator_params = PyTorchEstimator.estimator_params + NeuralNetworkMixin.estimator_params ` would be more explicit about the addition. About `self._variable`, I think we can leave it as is for now.

**Review by beat-buesser — APPROVED** [review]

**Pull request merged**